Heat_method_3/doc/Heat_method_3/Heat_method_3.txt

```diff
@@ -0,0 +1,249 @@
+        over the entire domain M, where id is the identity operator.
+
+The spatial discretization depends on the choice of discrete surface representation.
+For this package, we use triangle meshes exclusively.
+Let \f$ u \in \R^{|V|}\f$ specify a piecewise linear function on a
+triangulated surface with vertices \f$V\f$, edges \f$E\f$ and faces
+\f$F\f$.  A standard discretization of the Laplacian at vertex \f$i\f$
+is:
+
+\f$ {Lu}_i = \frac{1}{2A_i} \sum_{j}(cot \alpha_{ij} + cot \beta_{ij})(u_j-u_i)\f$ where \f$A_i\f$ is one third the area of all triangles incident on vertex \f$i\f$.
+
+The sum is taken over all of the neighboring vertices
+\f$j\f$. Further, \f$\alpha_{ij}\f$ and \f$\beta_{ij}\f$ are the
+angles opposing the corresponding edge \f$ij\f$. We express this
+operation via a matrix \f$L = M^{-1}L_c\f$ where \f$M \in
+R^{|V|x|V|}\f$ is a diagonal matrix containing the vertex areas and
+\f$L_c \in R^{|V|x|V|} \f$ is the cotan operator representing the
+remaining sum.
+
+From this, the symmetric positive-definite system
+\f$(M-tL_C)u = \delta_{\gamma}\f$ can be solved to find
+\f$u\f$ where \f$\delta_{\gamma}\f$ is the Kronecker delta over \f$\gamma\f$.
+
+Next, the gradient in a given triangle can be expressed as:
+
+\f$\nabla \cdot X = \frac{1}{2} \sum_j cot\theta_1 (e_1 \cdot X_j) + cot \theta_2 (e_2 \cdot X_j)\f$
+
+where the sum is taken over incident triangles \f$j\f$ each with a vector \f$X_j\f$,
+\f$e_1\f$ and \f$e_2\f$ are the two edge vectors of triangle \f$j\f$
+containing \f$i\f$ and \f$\theta_1\f$, \f$\theta_2\f$ are the opposing angles.
+
+Finally, let \f$b \in R^{|V|}\f$ be the integrated divergences of the normalized vector field X.
+Thus, solving the symmetric Poisson problem \f$ L_c \phi = b\f$ computes the final distance function.
+
+\subsection Subsection_HM_IDT_Definitions Intrinsic Delaunay Triangulation
+
+The standard discretization of the cotan Laplace operator uses the cotangents of the angles in the triangle mesh.
+The intrinsic Delaunay algorithm constructs an alternative triangulation of the same polyhedral surface, which
+in turn yields a different (typically more accurate) cotan Laplace operator.  Conceptually, the edges of the iDT
+still connect pairs of vertices from the original (input) surface, but are now allowed to be geodesic paths along
+the polyhedron and do not have to correspond to edges of the input triangulation.  These paths are not stored
+explicitly; instead, we simply keep track of their lengths as the triangulation is updated.  These lengths are
+sufficient to determine areas and angles of the intrinsic triangles, and in turn, the new cotan Laplace matrix.
+
+An edge of a mesh is locally Delaunay if the sum of opposite angles is no smaller than pi, or equivalently,
+if the cotangents of the opposing angles is nonnegative. A mesh is Delaunay if all of its edges are locally Delaunay.
```

Comment:
is nonnegative -> are non-negative

Reply:
fixed. I also changed nonzero to non-zero.